libc/utils/MPFRWrapper/MPFRUtils.h

```diff
@@ -293,7 +296,10 @@ constexpr bool is_valid_operation() {
       (op == Operation::Fma && internal::IsTernaryInput<InputType>::VALUE &&
        cpp::is_floating_point_v<
            typename internal::MakeScalarInput<InputType>::type> &&
-       cpp::is_floating_point_v<OutputType>);
+       cpp::is_floating_point_v<OutputType>) ||
+      (op == Operation::Fmul &&
+       !internal::AreMatchingBinaryInputAndBinaryOutput<InputType,
```

Comment:
it should be similar conditions to the `Operation::Fma` above, with `IsBinaryInput` instead of `IsTernaryInput`.